dhctl/pkg/kubernetes/actions/manifests/manifests.go

```diff
@@ -68,190 +74,234 @@ func GetDeckhouseDeployTime(deployment *appsv1.Deployment) time.Time {
+		deckhouseContainerEnv = append(deckhouseContainerEnv,
+			apiv1.EnvVar{
+				Name:  "KUBERNETES_DEPLOYED",
+				Value: params.DeployTime.Format(deployTimeEnvVarFormat),
 			},
-		}
+		)
+	}
 
-		volumeIndx := -1
+	if params.MasterNodeSelector {
+		deckhousePodTemplate.Spec.NodeSelector = map[string]string{"node-role.kubernetes.io/master": ""}
+	}
 
-		for i := range deployment.Spec.Template.Spec.Volumes {
-			if deployment.Spec.Template.Spec.Volumes[i].Name == deckhouseRegistryVolumeName {
-				volumeIndx = i
-				break
-			}
+	if params.IsSecureRegistry {
+		deckhousePodTemplate.Spec.ImagePullSecrets = []apiv1.LocalObjectReference{
+			{Name: "deckhouse-registry"},
 		}
 
-		if volumeIndx < 0 {
-			deployment.Spec.Template.Spec.Volumes = append(deployment.Spec.Template.Spec.Volumes, volumeToSet)
-		} else {
-			deployment.Spec.Template.Spec.Volumes[volumeIndx] = volumeToSet
+		var volumeFound bool
+		for _, volume := range deckhousePodTemplate.Spec.Volumes {
+			volumeFound = volumeFound || volume.Name == deckhouseRegistryVolumeName
 		}
 
-		// set volume mount
-
-		volumeMountToSet := apiv1.VolumeMount{
-			Name:      deckhouseRegistryVolumeName,
-			MountPath: "/etc/registrysecret",
-			ReadOnly:  true,
+		if !volumeFound {
+			deckhousePodTemplate.Spec.Volumes = append(deckhousePodTemplate.Spec.Volumes, apiv1.Volume{
+				Name: deckhouseRegistryVolumeName,
+				VolumeSource: apiv1.VolumeSource{
+					Secret: &apiv1.SecretVolumeSource{SecretName: deckhouseRegistrySecretName},
+				},
+			})
 		}
 
-		volumeMountIndx := -1
+		var volumeMountFound bool
+		for _, volumeMount := range deckhouseContainer.VolumeMounts {
+			volumeMountFound = volumeMountFound || volumeMount.Name == deckhouseRegistryVolumeName
+		}
 
-		for i := range deployment.Spec.Template.Spec.Containers[0].VolumeMounts {
-			if deployment.Spec.Template.Spec.Containers[0].VolumeMounts[i].Name == deckhouseRegistryVolumeName {
-				volumeMountIndx = i
-				break
-			}
+		if !volumeMountFound {
+			deckhouseContainer.VolumeMounts = append(deckhouseContainer.VolumeMounts, apiv1.VolumeMount{
+				Name:      deckhouseRegistryVolumeName,
+				MountPath: "/etc/registrysecret",
+				ReadOnly:  true,
+			})
 		}
+	}
 
-		if volumeMountIndx < 0 {
-			deployment.Spec.Template.Spec.Containers[0].VolumeMounts = append(deployment.Spec.Template.Spec.Containers[0].VolumeMounts, volumeMountToSet)
-		} else {
-			deployment.Spec.Template.Spec.Containers[0].VolumeMounts[volumeMountIndx] = volumeMountToSet
+	if params.KubeadmBootstrap && freshDeployment {
+		if !deployServiceHost {
+			deckhouseContainerEnv = append(deckhouseContainerEnv,
+				apiv1.EnvVar{
+					Name: deployServiceHostEnvVarName,
+					ValueFrom: &apiv1.EnvVarSource{
+						FieldRef: &apiv1.ObjectFieldSelector{APIVersion: "v1", FieldPath: "status.hostIP"},
+					},
+				},
+			)
+		}
+		if !deployServicePort {
+			deckhouseContainerEnv = append(deckhouseContainerEnv,
+				apiv1.EnvVar{
+					Name:  "KUBERNETES_SERVICE_PORT",
```

Comment:
I did not see it in last review, but we have const for name. We can use for it. 

https://github.com/deckhouse/deckhouse/pull/716/files#diff-49aea6002f1b4637e82d319b8c945b0eacbaee54c97975123db358619c35cca3R42